Sources/SwiftDocC/Indexing/Navigator/NavigatorIndex+Ext.swift

```diff
@@ -78,12 +78,18 @@ extension RenderNode {
     
     /// Returns a navigator title preferring the fragments inside the metadata, if applicable.
     func navigatorTitle() -> String? {
-        let pageType = navigatorPageType()
-        guard ![.framework, .class, .structure, .enumeration, .protocol, .typeAlias, .associatedType].contains(pageType) else { return metadata.title }
-        guard let fragments = metadata.fragments else { return metadata.title }
-        return fragments.map { token in
-            return token.text
-        }.joined()
+        let fragments: [DeclarationRenderSection.Token]?
+        
+        if identifier.sourceLanguage == .swift {
+            let pageType = navigatorPageType()
+            guard ![.framework, .class, .structure, .enumeration, .protocol, .typeAlias, .associatedType].contains(pageType) else { return metadata.title }
+            // FIXME: Use `metadata.navigatorTitle` (SR-15947).
+            fragments = metadata.fragments
+        } else {
+            fragments = metadata.navigatorTitle
+        }
+        
+        return fragments?.map(\.text).joined() ?? metadata.title
```

Comment:
What is the reason we need to look at the fragments for swift?

Reply:
I'm not entirely sure. I file https://bugs.swift.org/browse/SR-15947 for us to figure this out